identity_iota_core/src/rebased/client/full_client.rs

```diff
@@ -253,18 +259,42 @@ impl<S> IdentityClient<S> {
     Ok(budget)
   }
 
-  async fn get_coin_for_transaction(&self) -> Result<iota_sdk::rpc_types::Coin, Error> {
-    let coins = self
-      .coin_read_api()
-      .get_coins(self.sender_address(), None, None, None)
-      .await
-      .map_err(|err| Error::GasIssue(format!("could not get coins; {err}")))?;
+  async fn get_coin_for_transaction(&self) -> Result<Coin, Error> {
+    const LIMIT: usize = 10;
+    let mut cursor = None;
+
+    loop {
+      let coins = self
+        .coin_read_api()
+        .get_coins(self.sender_address(), None, cursor, Some(LIMIT))
+        .await?;
+
+      if coins.data.is_empty() {
+        return Err(Error::GasIssue("no coins found for address".to_string()));
+      }
+
+      if let Some(coin) = coins.data.into_iter().max_by_key(|coin| coin.balance) {
+        if coin.balance >= MINIMUM_BALANCE {
+          return Ok(coin);
+        }
+
+        return Err(Error::GasIssue(format!(
+          "highest balance coin ({}) is below minimum required balance of {}",
+          coin.balance, MINIMUM_BALANCE
+        )));
```

Comment:
Do we need this part here? ^^

Thinking about a case, where no coin object on the first page meetings the required `MINIMUM_BALANCE`. Then we wouldn't be returning with the `Ok` from line 278.

In this case, I'd expect the logic to page through the results until a page with a coin with enough funds is found. But with the current lines 281-284 we'd be canceling the paging early.

So, shouldn't the error in line 294 be enough, as it should be triggered after we'd paged through all results and still haven't found a matching coin? ^^